src/bot/bot_message.py

```diff
@@ -0,0 +1,52 @@
+"""Copyright by CookieDecide, Darkuuu
+Licensed under MIT License
+"""
+
+import discord
+from discord.ext import commands
+import logging
+import util.embed
+
+
+class BotMessage(commands.Cog):
+    """Collection of all Message commands and corresponding utility functions.
+
+    Attributes:
+        bot: Reference to discord bot.
+    """
+
+    def __init__(self, bot):
+        """Initializes the instance.
+
+        Args:
+            bot: Discord bot.
+        """
+        self.bot = bot
+
+    @commands.command(name="purge", help="Deletes the last x messages.")
+    async def purge(self, ctx: commands.Context, count: int):
+        """Deletes the last x messages.
+
+        Args:
+            ctx: Context of command invocation.
+            count: Number of messages to delete.
+        """
+        logging.info(f"Received purge request from user {ctx.author}")
+
+        deleted = 0
+        to_delete = count
+        while to_delete > 100:
```

Comment:
IMO code can be optimized:

Change to "while to_delete > 0". Then you can remove the lines with deleted_smaller_100 and the additional API call, since in the while loop all desired messages will be deleted.
Else you would have an additional call if you do for example >purge 142, 100 gets removed in first loop, 42 in second loop and then you give the purge API call a 0 after the loop, which is not necessary.

I know, only a small optimization, but why not

Reply:
I tried a different approach because i didnt like how cluttered it looked.
It should work now as intended and the number of purges per request is being logged.

Reply:
yep, this way also looks totally fine!